BugFix: The search bars at get directions don't replicate the start and end positions in text.

IdrisDrouiche opened the pull request:
## **Pull Request: Fix Issue Where Text Would Not Update Upon User Input in the Floating Search Bar**
This pull request implements a fix for the following bug.

**Fix**
User can now obtain feedback on their input, and see their input in the floating search bar.

NOTE: For code review focus on floatingSearchBar.js as all the other changes consist of automatic linter fixes.

**Test**
- [x] Unit Test

- [x] Manual Test

**Suporting Images**
Before (after location is entered no text is displayed in the search bar):
<img src="https://github.com/user-attachments/assets/6e6bb90d-5f42-4d1f-88fc-f261e01d753d" width ="200"><img src="https://github.com/user-attachments/assets/5256ae54-5675-4cd8-ab0c-1078f2bc1138" width ="200">

After (Displays text as feedback for the user when a location is inputed):
<img src="https://github.com/user-attachments/assets/c61c9c88-072b-41a2-83c2-6a54f0582b11" width ="200"><img src="https://github.com/user-attachments/assets/02087391-ae2c-42e7-bf1c-e212a1ef61cf" width ="200">

shanvinluo commented on ConcordiaMaps/components/FloatingSearchBar.js:
```diff
@@ -137,16 +137,15 @@ const FloatingSearchBar = ({ onPlaceSelect, placeholder }) => {
           <TouchableOpacity
             onPress={() => {
               setSearchQuery("");
+              setSelectedLocation("");
```
Setting the selected location using 'description' ensures clear context. This follows the "Rename Variable" refactoring from the catalog to improve clarity.

IdrisDrouiche replied:
Addressed this refactoring

shanvinluo reviewed (approved):
Except for a quick adding of adding a variable to ensure more clarity, the rest of the code seems good to me. Except for components related to the FloatingSearchBar, the rest seem to be coverage modifications or prettier modifications. The code runs on Android and I tested and verified that the functionalities are working as expected. I performed both unit and manual testing. For the unit tests, I ensured the input handling, location selection, and UI updates behaved correctly. During manual testing, I simulated various user inputs, confirmed the search query displayed accurately in the floating search bar, and verified that the selected location was correctly set and displayed. Everything functioned smoothly without issues.

LGTM! Approved!

shanvinluo commented:
Something we should address is putting the coverage folder in the gitignore since we've been getting merge conflicts ever since.

Atai-Askarov reviewed (approved):
I don't see anything that requires refactoring. The variables have appropriate names that describe the functionality of the variables. I don't see any duplications or functions that should be isolated to be relocated to another file. 
One thing tho, in order to make the review process more user friendly I would suggest to add the coverage and issue templates to git ignore. Only 1 file out of the 31 required actual review since the rest were auto generated.
Approved!

GitExcited commented on ConcordiaMaps/components/FloatingSearchBar.js:
```diff
@@ -19,7 +19,8 @@ const FloatingSearchBar = ({ onPlaceSelect, placeholder }) => {
   const [loading, setLoading] = useState(false);
-  const [selectedLocation, setSelectedLocation] = useState("");
+  // eslint-disable-next-line no-unused-vars
+  const [selectedLocation, setSelectedLocation] = useState(""); // reason: necessary fo running the appropriate tests.
```
Why is this necessary to run the tests? Maybe the test should be reworked if lint is flagging this as non-used variable

shanvinluo replied:
gitignore is updated on the develop branch though no? I think since those branches are older than when we added that.

IdrisDrouiche replied:
Good catch, it turns out there was an issue in the was the use state was set. The tests are working properly.

GitExcited reviewed (changes requested):
I left  acomment to review regarding the use non lint ignore. Additionally, the coverage files should be added to a .gitignore to avoid cluttering the commits.

IdrisDrouiche commented:
After addresssing the review comments these were the changes introduced:

- Coverage folder was removed from tracking and added to the gitignore as per our discussion.
- Addressed the refactoring
- Addressed the comment about linter affecting test.

shanvinluo reviewed (approved):
It seems that you implemented all the required changes and refactoring like we mentioned. Putting the coverage in the gitignore made it a lot easier to go through. I tested and verified that the functionalities are working as expected. I performed both unit and manual testing. For the unit tests, I ensured the input handling, location selection, and UI updates behaved correctly. During manual testing, I simulated various user inputs, confirmed the search query displayed accurately in the floating search bar, and verified that the selected location was correctly set and displayed. Everything functioned smoothly without issues. 

@GitExcited's comments have also been addressed so I see no reason not to merge. 

LGTM! Approved!
